modules/internal/DefaultRectangular.chpl

```diff
@@ -1968,10 +1971,41 @@ module DefaultRectangular {
     const Adata = _ddata_shift(A.eltType, A.theData, Aidx);
     const Bidx = B.getDataIndex(Blo);
     const Bdata = _ddata_shift(B.eltType, B.theData, Bidx);
-    _simpleTransferHelper(A, B, Adata, Bdata, len);
+
+    type t = A.eltType;
+    const elemsizeInBytes = if isNumericType(t) then numBytes(t)
+                            else c_sizeof(t).safeCast(int);
+    const doParallelAssign = len:int*elemsizeInBytes >= parallelAssignThreshold;
+
+    if doParallelAssign {
+      _simpleParallelTransferHelper(A, B, Adata, Bdata, len);
+    }
+    else{
+      _simpleTransferHelper(A, B, Adata, Bdata, len);
```

Comment:
Instead of having the 2 function calls here, can we just call `_simpleParallelTransferHelper` with a `maxTasks=1` arg or something? It shouldn't really matter since we're not inlining _simpleTransferHelper, but might be a little cleaner?

Reply:
How strongly do you feel about this?

I disagree with that: calling a "parallel" helper with a maxTasks=1 is less self-documenting than this I think. However, I don't feel strongly enough to argue much further than this :)

Reply:
I don't feel strongly at all and I buy the argument that this is easy to read. It jumped out to me because it kinda looks like the single-task specialization we had for parallel iterators, but in this case since things aren't being inlined it doesn't matter.

Reply:
Oh, I see why you suggested that better now. But yeah, this feels cleaner to me and doesn't have that impact, so I'm keeping it.